src/go/rpk/pkg/cli/cluster/partitions/list.go

```diff
@@ -248,3 +257,39 @@ func filterPartition(cPartitions []adminapi.ClusterPartition, partitions []int)
 	}
 	return
 }
+
+// filterBroker filters cPartition and returns a slice with only the
+// clusterPartitions with the same brokers present in the Replicas slice.
+func filterBroker(cPartitions []adminapi.ClusterPartition, brokers []int) (ret []adminapi.ClusterPartition) {
+	for _, p := range cPartitions {
+		rob := replicaOnBroker(p.Replicas, brokers)
```

Comment:
[question] So the filtering works in a way that we will only show partitions that have replicas in **all** the broker list that I pass, i.e:

- I have a partition `A` with a replica in broker 1
- I have a partition `B` with replicas in brokers 1 and 2

If I filter `--broker 1,2` will I only see partition `B` ?

Reply:
Yes, that's my intention. It's AND condition, not OR. Is it unclear?

Reply:
Awesome, it's clear. I just wanted to make sure. :+1: